overlord/state/taskrunner.go

```diff
@@ -258,8 +288,15 @@ func (r *TaskRunner) Ensure() {
 			continue
 		}
 
+		if blocked(t, running) {
```

Comment:
`if r.blocked != nil && r.blocked(t, running) {`

This would remove 4 lines out of the boilerplate above.